### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-input

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1479]
## Description
This task describes the Storybook implementation for the sd-input component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-input can be found here: [sd-input Notes](https://www.notion.so/matuzo/sd-input-01c5d524d0684f7ab8ba4890ca543361).

--------------------------------

## Comment
**Problem:** The [value example](https://solid-design-system.fe.union-investment.de/x.x.x/storybook/?path=/docs/components-sd-input--docs#value) has no accessible name and no visual label.
**Solution:** At least add an accessible name. Better: both.

**Problem:** The “Clear entry”-button is not accessible via the keyboard. That’s probably on purpose, but I don’t see why it shouldn’t be focusable.
**Solution:** Make sure the implementation is aligned with Shoelace as described here: https://github.com/shoelace-style/shoelace/issues/727#issuecomment-1098072111 and https://github.com/shoelace-style/shoelace/issues/727#issuecomment-1097239722

**Problem:** Invalid fields don't semantically indicate that they're invalid.
**Solution:** Put aria-invalid="true" on invalid fields.

**Problem:** The error message in invalid fields is not connected to the respective field.
**Solution:** Use aria-describedby, just like you're doing it with the help text.

```
<input aria-describedby="help-text invalid-message">
<div id="help-text">Help text</div>
<div id="invalid-message">Error message</div>
```

**Problem:** The buttons in the currency stepper have no accessible name.
**Solution:** Add a label to the icons.

**Problem:** Clicking the buttons in the currency stepper, screen reader users get no feedback on what happened.
**Solution:** Put `aria-live="polite"` on currency stepper input fields.

**Problem:** The “Show password”-button is not accessible via the keyboard.
**Solution:** Remove the `tabindex`-attribute.

**Problem:** Please note that `maxlength` doesn't indicate the maximum number of characters to screen readers. If there's no help text, it can be confusing.
**Solution:** Ensure the help text information properly describes what makes an input valid (`maxlength`, `minlength`)

<details>
<summary>
Notes for @miguelxnxn 
</summary>
  ## Notes
  - Instead of (or in addition to) changing the label, you could also toggle the `aria-pressed` attribute to indicate whether the password is visible.
  - Consider noting in the docs that the placeholder is not a replacement for the label but only an addition.
  - Disabled inputs: See [@sd-button](https://www.notion.so/sd-button-104f3f6e1bc0800b9f72d2a046316ec3?pvs=24). Everything I wrote there about disabling form elements applies here as well.
  
  ## Best Practices
  - Besides some rare exceptions, like in a search widget, always provide a visible label and don't rely on using placeholders alone.
</details>

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#labels-or-instructions

## Template Ticket
- [ ] Pick up the template ticket when working on the component story for A11y improvement, as it references the component and will be resolved once the component is made accessible:
#1520 

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by smfonseca** [comment]
> Problem: The “Clear entry”-button is not accessible via the keyboard. That’s probably on purpose, but I don’t see why it shouldn’t be focusable.
Solution: Make sure the implementation is aligned with Shoelace as described here: https://github.com/shoelace-style/shoelace/issues/727#issuecomment-1098072111 and https://github.com/shoelace-style/shoelace/issues/727#issuecomment-1097239722

> Problem: The “Show password”-button is not accessible via the keyboard.
Solution: Remove the tabindex-attribute.

@yoezlem @mariohamann the implementation of the `clear entry` and `show password` buttons follow the shoelace concept, therefore we do not deviate from it. I find this statement from the shoelace [maintainer](https://github.com/shoelace-style/shoelace/issues/727#issuecomment-1098072111) specially relevant:

> It feels dangerous, especially since Enter can trigger a form submit as well as activate the clear button. User might expect that tabbing will reach the next field and press it hoping for a submit, but instead their data will be erased. Perhaps this is why browsers omit it from the tab order.
